core/app_switcher/app_switcher.py

```diff
@@ -316,6 +317,20 @@ def switcher_focus_app(app: ui.App):
                 raise RuntimeError(f"Can't focus app: {app.name}")
             actions.sleep(0.1)
 
+    def switcher_focus_app_title(app: str, regex: str):
+        """Focus the window whose app name constains app and title matches regex"""
+        for window in ui.windows():
+            if (
+                not (window.hidden)
+                and (app in window.app.name or app == "*")
+                and window.title != ""
+            ):
+                # logging.warn(f'Checking Window: "{window.app.name}" window:"{window.title}" hidden: "{window.hidden}"')
+                if regex is None or re.search(regex, window.title):
```

Comment:
What's your usecase for passing a regex here?

Reply:
I currently have two commands that utilize an actual regex vs a plain string.

`focus outlook: user.switcher_focus_app_title("Outlook", "(Inbox|Calendar)")
focus studio: user.switcher_focus_app_title("Microsoft Visual Studio", "^(?:(?!Application Management).)*$")`

Reply:
I think this seems reasonable given the use cases. @auscompgeek any objection or you think it's ok?

Reply:
I suppose the use-case makes sense. I'm not sure we should make this the primary way, or indeed the only way, of focusing by title, as it'd be prone to user error – it's unclear from the action name that it takes a regex.